docs/architecture/ADR-004-reserves.md

```diff
@@ -0,0 +1,129 @@
+# ADR 004: Accumulating reserves from borrow interest
+
+## Changelog
+
+- October 14, 2021: Initial Draft (@toteki)
+
+## Status
+
+Proposed
+
+## Context
+
+Borrow positions on Umee accrue interest over time.
+When interest accrues, the sum of all assets owed by all users increases for each borrowed token denomination, and the amount of that increase serves to benefit lenders (by increasing the total amount of assets owed by the Umee system to lenders), and also to increase the amount of base assets the Umee system holds in reserve.
+
+The mechanism by which interest is calculated, and then split between incentivizing lenders as per [ADR-001](./ADR-001-interest-stream.md) and reserves as defined in this ADR, will follow.
+
+## Alternative Approaches
+
+- Setting aside a separate account (or even separate module account, for example by making `x/reserve`) would separate reserve balances from the lending pool.
+
+## Decision
+
+The `x/leverage` module already has a module account capable of storing base asset types, and it currently does so for assets in the lending pool.
+
+Adding reserve functionality should be as simple as using the existing module account to store reserve assets, and tracking (for each stored denomination) the amount which cannot be moved from the module acount (by means of a borrower's Borrow or a lender's Withdraw) because it is reserved.
+
+The aforementioned restrictions on borrows and withdrawals must then be implemented.
+
+Additionally, a governance parameter must be kept which specifies the portion of borrow interest that will be funnelled into reserves. This parameter will either be decided per token, or as a single parameter for all asset types.
+
+> TODO: Decide on the above.
```

Comment:
@brentxu how does Compound do this? Is it per token or total?

@toteki we should name the variable/param in this ADR.

Reply:
Now named `ReserveFactor` to match what we see Compound doing. Theirs appears to be per-token as well.